symfony/extracting-translations.rst

```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
       configs:
         app:
           external_translations_dir: ["%kernel.root_dir%/vendor/friendsofsymfony/user-bundle/Resources/translations"]
+
+
+Create custom extractor
+-----------------------
+
+Example method whose argument we want to translate 
+
+.. code-block:: php
+
+    $this->logger->addMessage("text");
+
+
+Example extractor class that we use to create translations
+
+.. code-block::	php
+
+    <?php
+
+    namespace App\Extractor;
+
+    use PhpParser\Node;
+    use PhpParser\NodeVisitor;
+    use Translation\Extractor\Visitor\Php\BasePHPVisitor;
+    
+    final class MyCustomExtractor extends BasePHPVisitor implements NodeVisitor 
+    {
+           
+        /**
+         * {@inheritdoc}
+        */
+        public function beforeTraverse(array $nodes): ?Node
+        {
+            return null;
+        }
+
+        /**
+        * {@inheritdoc}
+        */
+        public function enterNode(Node $node): ?Node
+        {
+            if (!$node instanceof Node\Expr\MethodCall) {
+                return null;
+            }
+
+            if (!is_string($node->name) && !$node->name instanceof Node\Identifier) {
+                return null;
+            }
+            
+            $name = (string) $node->name;
+
+            //This "if" check that we have method which interests us
+            if ($name !== "addMessage") {
+                return null;
+            }
+
+            $caller = $node->var;
+            $callerName = isset($caller->name) ? (string) $caller->name : '';
+           
+            //This "if" check that we have xxx->logger->addMessage() 
+            if ($name === 'addMessage' && $callerName === 'logger' && $caller instanceof Node\Expr\MethodCall) {
```

Comment:
probably `$name === 'addMessage'` check is redundant here? As we already make sure about this fact in the previous if, i.e. in `if ($name !== "addMessage") { return null; }`